### ISSUE TITLE
Assets not built with link `/src/assets`

### ISSUE THREAD

**Issue by WilfSilver** [withastro/astro#10634]
### Astro Info

```block
Astro                    v4.0.3
Node                     v21.5.0
System                   Linux (x64)
Package Manager          yarn
Output                   static
Adapter                  none
Integrations             @astrojs/starlight
                         @astrojs/react
```

### If this issue only occurs in one browser, which browser is a problem?

All

### Describe the Bug

I have an `md` file referencing an image/gif:

```md
...
![my test image](/src/assets/image.gif)
...
```

If I run `astro dev`, it displays the image fine, however running `astro build`, the image does not show on the website (in place it just has the alt text)

From looking at the files, it does not transfer/build the images into the `dist` directory.

If I change the link to a relative one e.g. `../../assets/image.gif` it works absolutely fine

### What's the expected result?

Consistency between `astro dev` and `astro build` when using `/src/assets` (so basically the image appears)

### Link to Minimal Reproducible Example

https://github.com/WilfSilver/astro-asset-link-issue

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by WilfSilver** [comment]
After further investigation, the issue seems to be that the images are actually displaying in `astro dev`.

I should be using the `import.meta.env.ASSETS_PREFIX` or via the `path` option in the typescript config explained [here](https://docs.astro.build/en/guides/aliases/)

Therefore, the expected result should be that `astro dev` does not display the images/prints out an error saying the images can't be found.

**Comment by bluwy** [comment]
Vite supports root-relative paths as it's advertised as a glorified filesystem server that serves files from your project root. There's a few explorations in the past to make this more strict, e.g. only allowing behind `/@something/src/assets/...` only, but that will be a really big breaking change.

I _think_ we can support this in Astro if we want to since we call `this.resolve` here:

https://github.com/withastro/astro/blob/f226cb65d166779a154fca07904a1cab662f6ac8/packages/astro/src/vite-plugin-markdown/index.ts#L80-L81

And it can resolve root-relative paths and public paths, and do some additional checks there. But I'm also fine if we want a blanket "starts with / is never handled": https://github.com/withastro/astro/blob/f226cb65d166779a154fca07904a1cab662f6ac8/packages/markdown/remark/src/remark-collect-images.ts#L28-L31

**Comment by matthewp** [comment]
I don't think we should try and support this, otherwise people will expect it in other places as well.

**Comment by ematipico** [comment]
Yeah, I agree with Matthew here. I would keep the behaviour as simple as possible and clarify the user expectations.

**Comment by matthewp** [comment]
After discussing this we think this might be reasonable to support and not be too difficult to implement.

**Comment by DJOCKER-FACE** [comment]
in the meantime is there any way to have the images display normally on the build?

**Comment by bluwy** [comment]
You can use relative paths, or use [aliases](https://docs.astro.build/en/guides/aliases/) like `@assets/my-image.png`

**Comment by DJOCKER-FACE** [comment]
@bluwy those didn't work, relative paths get passed as a string of text for some reason when used in .md and importing through assets directly doesn't work for build.

So this is how i kinda fixed this, i had to add those images on public/images, have to update astro(if not running latest) and then import them.

**Comment by WilfSilver** [comment]
> @bluwy those didn't work, relative paths get passed as a string of text for some reason when used in .md and importing through assets directly doesn't work for build.

Could you share the code for this? Relative links and aliases were working in my original project (and the example code - relative links are used as an example of how it should be working)
